### PR TITLE
metadata and warnings in response

### PR DESCRIPTION
**Pull request by swapnilp0x00**
- updated makeASCTBData function. it will now return data, metadata, and warnings.
- Added build metadata function to extract metadata from CSV arrays.

- updated responses for the following endpoints.
GET /v2/:sheetid/:gid
GET /v2/playground
POST /v2/playground

- updated API spec document.

### CODE REVIEW

File: asctb-api/src/index.ts [hunk 1]
```diff
@@ -13,11 +12,10 @@ import { setupPlaygroundRoutes } from './routes/playground';
 import { setupStaticPageRoutes } from './routes/static-pages';
 import { routeCache } from './utils/route-caching';
 
-
 export const app = express();
 app.use(cors());
-app.use(bodyParser.urlencoded({ extended: false }));
-app.use(bodyParser.json());
+app.use(express.urlencoded({ extended: false }));
```
Comment (bherr2) [inline asctb-api/src/index.ts:17]: Are you sure we aren't supposed to use bodyParser? I thought they had moved some stuff out of express and you were supposed to use them fro separate modules.

File: asctb-api/src/routes/csv.ts [hunk 2]
```diff
@@ -34,9 +33,10 @@ export function setupCSVRoutes(app: Express): void {
           const response = await axios.get(parsedUrl);
           csvData = response.data;
 
-          const data = papa.parse(response.data, { skipEmptyLines: 'greedy' }).data;
+          const { data } = papa.parse<string[]>(response.data, { skipEmptyLines: 'greedy' });
           parsedCsvData = data;
-          return makeASCTBData(data);
+          const asctbData = makeASCTBData(data);
+          return asctbData.data;
```
Comment (bherr2) [inline asctb-api/src/routes/csv.ts:39]: This loses all metadata and warnings ...

File: asctb-api/src/index.ts (hunk 1 above)
Comment (swapnilp0x00) [reply asctb-api/src/index.ts:17]: Yes, they were removed from express in version 4.0.0 but readded in 4.16.0 https://github.com/expressjs/express/releases/tag/4.16.0

We are using express 4.17.0 and it was giving me a warning as  `deprecated body-parser`
https://stackoverflow.com/questions/47232187/express-json-vs-bodyparser-json

Comment (bherr2) [reply asctb-api/src/index.ts:17]: Ahh ok :+1:

File: asctb-api/src/functions/api.functions.ts [hunk 3]
```diff
@@ -1,7 +1,11 @@
-/* tslint:disable:variable-name */
-import { arrayNameMap, createObject, HEADER_FIRST_COLUMN, objectFieldMap, Row } from '../models/api.model';
+import { arrayNameMap, createObject, DELIMETER, HEADER_FIRST_COLUMN, metadataNameMap, objectFieldMap, Row } from '../models/api.model';
 import { fixOntologyId } from './lookup.functions';
 
+export interface ASCTBData {
+  data: Row[],
```
Comment (bherr2) [inline asctb-api/src/functions/api.functions.ts:5]: Those should end with a semi-colon not a comma

**Review by bherr2 — CHANGES_REQUESTED** [review]

**Review by bherr2 — APPROVED** [review]

**Pull request merged**